docs/REFERENCE.md

```diff
@@ -481,6 +542,46 @@ const okOfNumber5 = originOk.mapAnyErr(err => 123);
 
 ---
 
+### UnknownError
+`UnknownError` is a special error class used to wrap unexpected errors that are thrown within the `map`, `mapErr`, `mapAnyErr`, or `Result.from` context. It has a `cause` property that contains the original error.
```

Comment:
I don't know where exactly but I think there should be said something like that:

`UnknownError` is the only error which you don't need to handle before using `.promise()`. The reason is simple even if you do `mapErr` with `UnknownError` as first parameter your code could throw an error in handler function anyway so it can create a chain of `mapErr` behaviours which is against library idea.